test/SlowTests/Issues/RavenDB_12816.cs

```diff
@@ -0,0 +1,391 @@
+        public RavenDB_12816(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public void CanSendFacetedRawQuery()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var index = new CameraCost();
+                index.Execute(store);
+
+                using (var commands = store.Commands())
+                {
+                    for (var i = 0; i < 10; i++)
+                    {
+                        commands.Put(
+                            "cameras/" + i,
+                            null,
+                            new Camera
+                            {
+                                Id = "cameras/" + i,
+                                Manufacturer = i % 2 == 0 ? "Manufacturer1" : "Manufacturer2",
+                                Cost = i * 100D,
+                                Megapixels = i * 1D
+                            },
+                            new Dictionary<string, object> { { "@collection", "Cameras" } });
+                    }
+
+
+                    WaitForIndexing(store);
+
+                    var facets = new List<Facet>
+                    {
+                        new Facet
+                        {
+                            FieldName = "Manufacturer"
+                        }
+                    };
+
+                    var rangeFacets = new List<RangeFacet>
+                    {
+                        new RangeFacet
+                        {
+                            Ranges =
+                            {
+                                "Cost <= 200",
+                                "Cost >= 300 and Cost <= 400",
+                                "Cost >= 500 and Cost <= 600",
+                                "Cost >= 700 and Cost <= 800",
+                                "Cost >= 900"
+                            }
+                        },
+                        new RangeFacet
+                        {
+                            Ranges =
+                            {
+                                "Megapixels <= 3",
+                                "Megapixels >= 4 and Megapixels <= 7",
+                                "Megapixels >= 8 and Megapixels <= 10",
+                                "Megapixels >= 11",
+                            }
+                        }
+                    };
+
+
+                    commands.Put(
+                        "facets/CameraFacets",
+                        null,
+                        new FacetSetup { Id = "facets/CameraFacets", Facets = facets, RangeFacets = rangeFacets },
+                        null);
+
+                    WaitForIndexing(store);
+
+                    using (var session = store.OpenSession())
+                    {
+                        var facetResults = session
+                            .Advanced
+                            .RawQuery<Camera>("from index 'CameraCost' select facet(id('facets/CameraFacets'))")
```

Comment:
Need to give good error if we are failing to parse the query because they called ToList() here.

Reply:
done